### DISCUSSION TITLE
Missing p2 dependencies in maven dependency:tree output

### DISCUSSION THREAD

**Discussion by eNBeWe** [eclipse-tycho/tycho#3882] [Q&A]
With the new target platform mechanism in Tycho 4, the output from `dependency:tree` is missing all P2 dependencies, if run only as a standalone target.
With the previous resolution mechanism, the tree is complete.

We have some tooling that uses dependency:tree under-the-hood and we can't change the executed goals, but this makes it hard to use the new approach.

**Comment by laeubi** [comment]
For legacy use please try `-Dtycho.target.eager=true` beside that your tooling seem to make wrong assumptions if it assumes that a `dependency:tree` will discover any dynamic dependencies injected by extensions.

If you want all P2 dependencies there is also https://tycho.eclipseprojects.io/doc/latest/tycho-p2-plugin/dependency-tree-mojo.html
